app/services/UserServiceImpl.js

```diff
@@ -29,7 +29,30 @@ module.exports = function(userDao) {
     });
   };
 
+  const signin = function(userData) {
+    return new Promise((resolve, reject) => {
+      _userDao.signin(userData)
+        .then((user) => {
+          comparePassword(userData.password, user.password)
```

Comment:
This needs to be returned in order to continue the promise chain.

```javascript
.then((user) => {
  return comparePassword(userData.password, user.password);
})
```

But this method will probably need to be changed because `comparePassword` should probably return a `boolean`, and can then be renamed to something like `isMatchingPassword` to better convey what the method is intended to do.